infer/src/pulse/PulseFormula.ml

```diff
@@ -2966,10 +3047,15 @@ module Formula = struct
 
     val normalize_atom : t -> Atom.t -> Atom.t list SatUnsat.t
 
-    val and_normalized_atoms : t * new_eqs -> Atom.t list -> (t * new_eqs) SatUnsat.t
+    val and_normalized_atoms :
+         t * new_eqs
+      -> Atom.t list
+      -> orig_atom:Atom.t list
+      -> add_term:bool
+      -> (t * new_eqs) SatUnsat.t
     (** use with the result of {!normalize_atom} in place of {!and_atom} *)
 
-    val and_atom : Atom.t -> t * new_eqs -> (t * new_eqs) SatUnsat.t
+    val and_atom : Atom.t -> t * new_eqs -> bool -> (t * new_eqs) SatUnsat.t
```

Comment:
please name boolean and integer arguments. Tthis could be made an optional argument to avoid changing too much client code (but also optional arguments are evil as they are easy to forget so... your call)
```suggestion
    val and_atom : add_raw_term:bool -> Atom.t -> t * new_eqs -> (t * new_eqs) SatUnsat.t
```